application/core/LimeMailer.php

```diff
@@ -767,17 +767,19 @@ public static function validateAddress($address, $patternselect = null)
         if (null === $patternselect) {
             $patternselect = static::$validator;
         }
-        if ($patternselect != 'php-idna') {
-            return parent::validateAddress($address, $patternselect);
-        }
-        require_once(APPPATH . 'third_party/idna-convert/idna_convert.class.php');
-        $oIdnConverter = new idna_convert();
-        $sEmailAddress = $oIdnConverter->encode($address);
-        $bResult = filter_var($sEmailAddress, FILTER_VALIDATE_EMAIL);
-        if ($bResult !== false) {
-            return true;
-        }
-        return false;
+        if ($patternselect == 'php-idna') {
+            /**
+             * PHPMailer has support for IDN, but it relies on `intl` and `mbstring`.
+             * If we use 'idna_convert' as we did, we are not ensuring that PHPMailer
+             * can handle the address later. So instead of using 'idna_convert' we use
+             * PHPMailer's punyencodeAddress() method.
+             */
+            $mailer = new \PHPMailer\PHPMailer\PHPMailer(); // Can't use an instance of LimeMailer because of a recursion problem
```

Comment:
No way to use parent::punyencodeAddress ? or this->punyencodeAddress ?

Reply:
Nope, there is a recursion problem

Reply:
Strange  ?